packages/workspace/src/command-line/affected.ts

```diff
@@ -45,8 +45,14 @@ export async function affected(
     );
   }
   const projects = parsedArgs.all ? projectGraph.nodes : affectedGraph.nodes;
-  const env = readEnvironment(nxArgs.target, projects);
-  const affectedProjects = Object.values(projects)
+  const projectsNotExcluded = Object.keys(projects)
+    .filter((key) => !parsedArgs.exclude.includes(key))
+    .reduce((p, key) => {
+      p[key] = projects[key];
+      return p;
+    }, {} as Record<string, ProjectGraphNode>);
+  const env = readEnvironment(nxArgs.target, projectsNotExcluded);
+  const affectedProjects = Object.values(projectsNotExcluded)
     .filter((n) => !parsedArgs.exclude.includes(n.name))
```

Comment:
I think you can remove this line now cause the filtering is happening above

Reply:
done